Divine/CLI/CommandLineActions.cs

```diff
@@ -0,0 +1,150 @@
+            {
+                PackageVersion = CommandLineArguments.GetPackageVersion(args.PackageVersion);
+                CommandLineLogger.LogDebug($"Using package version: {PackageVersion}");
+            }
+
+            if (graphicsActions.Any(args.Action.Contains))
+            {
+                GraphicsOptions = CommandLineArguments.GetGraphicsOptions(args.Options);
+                CommandLineLogger.LogDebug($"Using graphics options: {GraphicsOptions}");
+
+                if (GraphicsOptions["conform"])
+                {
+                    ConformPath = TryToValidatePath(args.ConformPath);
+                }
+            }
+
+            SourcePath = TryToValidatePath(args.Source);
+            DestinationPath = TryToValidatePath(args.Destination);
+        }
+
+        private static void Process(CommandLineArguments args)
+        {
+            switch (args.Action)
+            {
+                case "create-package":
+                    CommandLinePackageProcessor.Create();
+                    break;
+                case "extract-package":
+                    CommandLinePackageProcessor.Extract();
+                    break;
+                case "convert-model":
+                    CommandLineGraphicsProcessor.UpdateExporterSettings();
+                    CommandLineGraphicsProcessor.Convert();
+                    break;
+                case "convert-resource":
+                    CommandLineDataProcessor.Convert();
+                    break;
+                case "extract-packages":
+                    CommandLinePackageProcessor.BatchExtract();
+                    break;
+                case "convert-models":
+                    CommandLineGraphicsProcessor.BatchConvert();
+                    break;
+                case "convert-resources":
+                    CommandLineDataProcessor.BatchConvert();
+                    break;
+            }
+        }
+
+        public static string TryToValidatePath(string path)
+        {
+            const int MAX_PATH = 248;
+
+            CommandLineLogger.LogDebug($"Using path: {path}");
+
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                CommandLineLogger.LogFatal($"Cannot parse path from input: {path}");
+            }
+
+            Uri uri = null;
+            try
+            {
+                Uri.TryCreate(path, UriKind.RelativeOrAbsolute, out uri);
+            }
+            catch (InvalidOperationException)
+            {
+                CommandLineLogger.LogFatal($"Cannot proceed without absolute path [E1]: {path}");
+            }
+
+            if (uri != null && (!Path.IsPathRooted(path) || !uri.IsFile))
+            {
+                CommandLineLogger.LogFatal($"Cannot proceed without absolute path [E2]: {path}");
+            }
+
+            // ReSharper disable once AssignNullToNotNullAttribute
+            path = Path.GetFullPath(path);
+
+            if (path.Length > MAX_PATH)

```

Comment:
This one is probably unnecessary, as we're now on v4.6.2 where MAX_PATH is much larger. (Not sure though?)

Reply:
CMD doesn't support long paths yet.

Reply:
Damn, I totally did not see that. Then this check shall stay.